pkg/kubectl/cmd/cp_test.go

```diff
@@ -497,3 +500,72 @@ func TestClean(t *testing.T) {
 		}
 	}
 }
+
+func TestCopyToPod(t *testing.T) {
+	tf := cmdtesting.NewTestFactory()
+	tf.Namespace = "test"
+	tf.Client = &fake.RESTClient{}
+
+	tf.ClientConfigVal = defaultClientConfig()
+	buf := bytes.NewBuffer([]byte{})
+	errBuf := bytes.NewBuffer([]byte{})
+	cmd := NewCmdCp(tf, buf, errBuf)
+
+	var srcFile string
+	var err error
+	if srcFile, err = ioutil.TempDir("", "test"); err != nil {
+		t.Errorf("unexpected error: %v", err)
+		t.FailNow()
+	}
+
+	defer func() {
+		if err := os.RemoveAll(srcFile); err != nil {
+			t.Errorf("Unexpected error cleaning up: %v", err)
+		}
+	}()
+
+	tests := map[string]struct {
+		dest        string
+		expectedErr bool
+	}{
+		"copy to regular file": {
+			dest:        "/tmp/",
+			expectedErr: false,
+		},
+		"copy to root": {
+			dest:        "/",
+			expectedErr: false,
+		},
+		"copy to empty file address": {
+			dest:        "",
+			expectedErr: true,
+		},
+	}
+
+	for name, test := range tests {
+		src := fileSpec{
+			File: srcFile,
+		}
+		dest := fileSpec{
+			PodNamespace: "pod-ns",
+			PodName:      "pod-name",
+			File:         test.dest,
+		}
+		t.Run(name, func(t *testing.T) {
+			err = copyToPod(tf, cmd, buf, errBuf, src, dest)
+			if test.expectedErr && isConnectionRefusedError(err) {
+				t.Errorf("%s: expect error happens, but validate passes.", name)
+			}
+			if !test.expectedErr && !isConnectionRefusedError(err) {
+				t.Errorf("%s: unexpected error: %v", name, err)
+			}
+		})
+	}
+}
+
+func isConnectionRefusedError(err error) bool {
```

Comment:
This is bad and might be implementation dependent. It's fine just to test the failures you're adding without this going further.

Reply:
This is still not addressed.

Reply:
I thought you meat that the client would send a request to 127.0.0.1 and this may cause dependent problems.
So it seems we need not to restrict the failure into a narrow bound.